src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -133,14 +133,29 @@ def register_builtin_func(
     def domain(self) -> common.Domain:
         return self._domain
 
+    @property
+    def __gt_dims__(self) -> common.DimsT:
+        return self._domain.dims
+
+    @property
+    def __gt_origin__(self) -> tuple[int, ...]:
+        return tuple(-r.start for _, r in self._domain)
+
     @property
     def ndarray(self) -> core_defs.NDArrayObject:
         return self._ndarray
 
+    def __array__(self) -> np.ndarray:
+        return np.asarray(self._ndarray)
+
     @property
     def value_type(self) -> type[core_defs.ScalarT]:
         return self._value_type
```

Comment:
```suggestion
    @property
    def value_type(self) -> type[core_defs.ScalarT]:
        return self.dtype.type
```
in `common.Field` or remove